spinn_machine/virtual_machine.py

```diff
@@ -212,74 +104,56 @@ class _VirtualMachine(object):
-            with_monitors=True, sdram_per_chip=SDRAM.DEFAULT_SDRAM_BYTES,
+            self, width, height, n_cpus_per_chip=None,
+            sdram_per_chip=SDRAM.DEFAULT_SDRAM_BYTES,
             down_chips=None, down_cores=None, down_links=None,
             router_entries_per_chip=Router.ROUTER_DEFAULT_AVAILABLE_ENTRIES,
             validate=True):
 
+        if n_cpus_per_chip is None:
+            n_cpus_per_chip = Machine.max_cores_per_chip()
         self._n_router_entries_per_router = router_entries_per_chip
 
-        if down_chips is None:
-            down_chips = []
-
-        # Verify the machine
-        # Check for not enough info or out of range
-        _verify_basic_sanity(version, width, height)
-
-        # Version 2/3
-        if version in Machine.BOARD_VERSION_FOR_4_CHIPS:
-            width, height = _verify_4_chip_board(
-                version, width, height, with_wrap_arounds)
-            self._machine = machine_from_size(
-                width, height, origin=self.ORIGIN)
-        # Version 4/5
-        elif version in Machine.BOARD_VERSION_FOR_48_CHIPS:
-            width, height = _verify_48_chip_board(
-                version, width, height, with_wrap_arounds)
-            self._machine = machine_from_size(
-                width, height, origin=self.ORIGIN)
-        # Autodetect
-        elif version is None:
-            _verify_width_height(width, height)
-            if with_wrap_arounds is None:
-                self._machine = machine_from_size(
-                    width, height, origin=self.ORIGIN)
-            elif with_wrap_arounds:
-                self._machine = FullWrapMachine(
-                    width, height, origin=self.ORIGIN)
-            else:
-                self._machine = NoWrapMachine(
-                    width, height, origin=self.ORIGIN)
-        else:
-            raise SpinnMachineInvalidParameterException(
-                "version",
-                version,
-                "The only supported version numbers are 2, 3, 4, 5")
+        _verify_width_height(width, height)
+        self._machine = machine_from_size(width, height, origin=self.ORIGIN)
 
         # Store the details
         self._sdram_per_chip = sdram_per_chip
-        if with_monitors:
-            self._with_monitors = 1
-            self._weird_processor = False
-        else:
-            self._with_monitors = 0
-            self._weird_processor = True
         self._n_cpus_per_chip = n_cpus_per_chip
-        if n_cpus_per_chip != Machine.MAX_CORES_PER_CHIP:
-            self._weird_processor = True
 
         # Store the down items
-        self._down_cores = defaultdict(set)
+        unused_chips = []
+        if down_chips is not None:
+            for down_chip in down_chips:
+                if isinstance(down_chip, IgnoreChip):
+                    if down_chip.ip_address is None:
+                        unused_chips.append((down_chip.x, down_chip.y))
```

Comment:
Is it worth warning the user when the down chip is going to be ignored?  Should it be ignored or used in this case?  I think the VM has virtual IP addresses too...

Reply:
The virtual chips are given ip addresses 127.0.x.y if they are a local 0,0.

Is there a use case for supporting ignores in the 127 range?

My thinking here is if there are ignores for specific phyisical  boards in the cfg to ignore them on virtual machines,

Reply:
I think you are right to ignore specific down chips in virtual machines.  I would expect a user to comment out the down-chips that they have used in that case, but if they do want one on a virtual machine, it could be useful to be specified.  I guess the use of an IP address is a bit odd here given that they are automatically generated; in which case it might be better to raise and exception here to make sure the user knows what they are doing.

Reply:
Currently (both in master and this pr) the maxMachineGenerators are not passed any ignore.

So it would be possible to raise an exception if there are ip addresses in the ignores and the user selects virtual.  However my feeling is that virtual is for testing and I don't feel we need to force the users to change their cfg between testing and a real run.

Reply:
OK, seems reasonable.  I guess users that have specific boards in their config wouldn't expect this to work with the VM in that case.